### PR TITLE
33: Introduce securebanking-common-domain module

### PR DESCRIPTION
**Pull request by matt-wills**
### securebanking-common-domain module

- Added FRDomesticConsent, which represents the object that will be returned from IDM (there are many more "consent" objects to add, but we are only starting with domestic payments).
- Added corresponding FR "data" classes for all equivalent OB model objects (easiest just to move these in one go as no editing was required).

Issue: https://github.com/SecureBankingAcceleratorToolkit/SecureBankingAcceleratorToolkit/issues/33

### CODE REVIEW

File: securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.Date;
+
+/**
+ * Represents a Domestic Payment Consent, which will be provided by the IDM.
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class FRDomesticConsent implements PaymentConsent {
```
Comment (matt-wills) [inline securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java:37]: Previously this was a `@Document`, but it is no longer being stored in the simulator's Mongo instance. This represents the consent DTO that should be retrieved from the IDM. It is required when saving a payment (e.g. to check the consent exists and update its status).

I'm not yet sure which of these fields are required, so I've kept them all for now - something for the team to discuss.

Note that this class is in `com.forgerock.securebanking.common.openbanking.domain.payment`. All other payment consent objects should be added here (e.g. scheduled payment consent etc). The corresponding "data" classes (e.g. `FRWriteDomesticConsent` in this class) sit one level below in `com.forgerock.securebanking.common.openbanking.domain.payment.data`. The same approach should be used for accounts, events etc.

Comment (jorgesanchezperez) [reply securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java:37]: This domain (data model) will be the standard of secure banking accelerator or it's a specific domain for open banking uk, I mean if this domain corresponds to a specific domain we must be change the name of the module to `common-uk-domain` or something like that, isn't?

**Review by jorgesanchezperez — DISMISSED** [review]

File: securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java (hunk 1 above)
Comment (matt-wills) [reply securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java:37]: I think you're right - they are OB specific.

I guess I shouldn't haven added OB specific error classes to `secure-common-error` either (it previously contained only `ErrorResponseTmp`, which aside from being an odd name, probably belongs in an OB project as well?).

Looks like I'll have to create 2 new modules.

File: pom.xml [hunk 2]
```diff
@@ -43,9 +43,10 @@
     </scm>
 
     <modules>
-        <module>securebanking-common-error</module>
         <module>securebanking-common-bom</module>
         <module>securebanking-common-openbanking-uk-datamodel</module>
+        <module>securebanking-common-openbanking-uk-error</module>
+        <module>securebanking-common-openbanking-uk-domain</module>
```
Comment (matt-wills) [inline pom.xml:49]: OB specific modules named more consistently.

File: securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java (hunk 1 above)
Comment (matt-wills) [reply securebanking-common-domain/src/main/java/com/forgerock/securebanking/common/openbanking/domain/payment/FRDomesticConsent.java]: Done.

File: pom.xml (hunk 2 above)
Comment (jorgesanchezperez) [reply pom.xml:49]: Perfect!

**Review by jorgesanchezperez — APPROVED** [review]
All ok!

**Pull request merged**